### DISCUSSION TITLE
How do I use baggage in traces?

### DISCUSSION THREAD

**Discussion by fuleow** [open-telemetry/opentelemetry-java#3113] [Q&A] [answered]
Hey all, we currently have some Java services manually instrumented with OpenTelemetry 1.0 exporting via the `OtlpGrpcSpanExporter` to OpenTelemetry collectors. The collectors then export the traces to Jaeger.

It is working well at the moment and we are looking into adding baggage to automatically propagate some cross cutting attributes like a customer's email. I did not see a lot of documentation or mention of baggage except for a few classes like the `W3CBaggagePropagator` and some test cases.

I was wondering if it's currently possible to have traces with baggage propagated to Jaeger. Are there any good examples or documentation available? If it's not, which components are still missing support? Thanks.

**Comment by jkwatson** [comment, marked as answer]
I'm not sure what you mean by "have traces with baggage propagated to Jaeger". Baggage is an independent concern from traces, and will be propagated if you have a propagator installed (such as the W3CBaggagePropagator you mentioned).

Are you looking to append baggage entries to spans? If so, you can easily write a SpanProcessor that does this when `onStart()` is called.

**Reply by fuleow** [reply]
Thanks @jkwatson. I thought baggage would be shown as a trace detail in the tracing backends like span attributes which is why I asked whether it would be propagated to Jaeger. From your response it sounds like baggage is only propagated between services and is not exported which does make sense.

Yes, the intent is to append these baggage items as span attributes. I will take a look at the SpanProcessor

**Reply by fuleow** [reply]
@jkwatson I'm still struggling with this. I have baggage propagation working and wrote my own `SpanProcessor`, but I am having a hard time adding my custom `SpanProcessor` to a service that's auto instrumented using the agent. I stumbled across #2022  by @iNikem and his [test repo](https://github.com/iNikem/opentelemetry-java-instrumentation-demo) is a really good reference of what I'd like to do (it has a `DemoSpanProcessor`). It's difficult to follow along the discussion in #2022 as there's been quite a bit of refactoring since then.

**Reply by jkwatson** [reply]
For that, I will have to defer over to the instrumentation folks. Can you open a discussion over there about how to add a custom span processor to the agent?
